app/src/main/java/org/simple/clinic/widgets/qrcodescanner/QrCodeScannerView.kt

```diff
@@ -85,16 +89,23 @@ class QrCodeScannerView(context: Context, attrs: AttributeSet) : FrameLayout(con
         codeScanner.errorCallback = null
       }
     }
+
+    return qrCodeScans
+        .withLatestFrom(qrCodeScansValve)
```

Comment:
Is there a reason we're doing this manually instead of using [valve](https://github.com/akarnokd/RxJavaExtensions#flowabletransformersvalve)?

Reply:
The valve, buffers events that are received when it is off and delivers it downstream when it is back on. What we wanted was to drop event when the valve is shut. There is a overloaded value with a buffer size, but it requires a non-zero value. Hence, this.